### PR TITLE
implement support of context path

### PR DESCRIPTION
**Pull request by brastak**
For #15

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#21](https://github.com/artipie/npm-proxy-adapter/issues/21) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/5245388d-9972-41bd-8cee-65127a6ae0f3); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/npm/proxy/http/AssetPath.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.npm.proxy.http;
+
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Asset path helper.
```
Comment (Vatavuk) [inline src/main/java/com/artipie/npm/proxy/http/AssetPath.java]: @brastak please expand the description.

File: src/main/java/com/artipie/npm/proxy/http/NpmPath.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public NpmPath(final String prefix) {
+        this.base = prefix;
+    }
+
+    /**
+     * Gets relative path from absolute.
+     * @param abspath Absolute path
+     * @return Relative path
+     */
+    public String value(final String abspath) {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/npm/proxy/http/NpmPath.java]: @brastak make this method final

File: src/main/java/com/artipie/npm/proxy/http/NpmPath.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                )
+            );
+        }
+    }
+
+    /**
+     * Gets base path prefix.
+     * @return Bas path prefix
+     */
+    public String prefix() {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/npm/proxy/http/NpmPath.java]: @brastak make this method final

File: src/main/java/com/artipie/npm/proxy/http/PackagePath.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.npm.proxy.http;
+
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Package path helper.
```
Comment (Vatavuk) [inline src/main/java/com/artipie/npm/proxy/http/PackagePath.java]: @brastak please expand the description

**Review by Vatavuk — CHANGES_REQUESTED** [review]
@brastak comments below

**Comment by brastak** [comment]
@Vatavuk please see update

**Review by Vatavuk — APPROVED** [review]
@brastak thx
@g4s8 good to merge

**Review by g4s8 — APPROVED** [review]

**Comment by 0crat** [comment]
Job was finished in 25 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**